### ISSUE TITLE
defferent output at exported model

### ISSUE THREAD

**Issue by kspook** [emedvedev/attention-ocr#135]
@emedvedev, in my case, it doesn't work when I run exported model. T.T
Do you have solution?

**Comment by emedvedev** [comment]
There's not much to go on, so could you maybe provide more information? I may not be able to help, since it's your own fork with significant changes, but maybe someone else would take a look.

Does your Korean fork include the latest master changes? What's the correct fork/branch to look at? How exactly are you exporting the model? Do you have different results _for the exact same image_ with `predict` and with the same model on the same stage of training when exported?  What are the exact results, ideally with full logs?

Unless your issue includes all that, it would be very hard to see where the issue comes from.

**Comment by kspook** [comment]
thank you for your response.

my information as belows;
https://github.com/kspook/aocrKR.git
checkpoint: https://drive.google.com/open?id=1tPHoE1gK-AUjYBqbSCFbSc6lRnBh7WI5
model :

**Comment by kspook** [comment]
In addition to test 1.jpeg the above posting, I will continue for 'shin'
it shows the same result.

1. for test
2019-06-08 13:17:19,422 root  INFO     Step 6 (0.095s). Accuracy: 16.67%, loss: 3.153957, perplexity: 23.4286, probability: 91.60%   0% (T vs 신) 

2. predict
(py37) kspook@ml004:~/aocrKR$ python aocr36/__main__.py predict dataset/test-img3/shin.jpeg
2019-06-08 13:18:12.320031: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-06-08 13:18:12.326873: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2397220000 Hz
2019-06-08 13:18:12.327057: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x556c9f033c20 executing computations on platform Host. Devices:
2019-06-08 13:18:12.327136: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
2019-06-08 13:18:12,327 root  INFO     phase: predict
2019-06-08 13:18:12,328 root  INFO     model_dir: checkpoints
2019-06-08 13:18:12,328 root  INFO     load_model: True
2019-06-08 13:18:12,328 root  INFO     output_dir: results
2019-06-08 13:18:12,328 root  INFO     steps_per_checkpoint: 0
2019-06-08 13:18:12,328 root  INFO     batch_size: 1
2019-06-08 13:18:12,328 root  INFO     learning_rate: 1.000000
2019-06-08 13:18:12,329 root  INFO     reg_val: 0
2019-06-08 13:18:12,329 root  INFO     max_gradient_norm: 5.000000
2019-06-08 13:18:12,329 root  INFO     clip_gradients: True
2019-06-08 13:18:12,329 root  INFO     max_image_width 260.000000
2019-06-08 13:18:12,329 root  INFO     max_prediction_length 18.000000
2019-06-08 13:18:12,329 root  INFO     channels: 1
2019-06-08 13:18:12,329 root  INFO     target_embedding_size: 10.000000
2019-06-08 13:18:12,330 root  INFO     attn_num_hidden: 128
2019-06-08 13:18:12,332 root  INFO     attn_num_layers: 2
2019-06-08 13:18:12,333 root  INFO     visualize: False
filename, dataset/test-img3/shin.jpeg
2019-06-08 13:18:18,207 root  INFO     Result: OK. 0.92 T


3. curl.
(base) kspook@ml004:~$ curl -X POST \
>   http://localhost:9002/v1/models/aocr:predict \
>   -H 'cache-control: no-cache' \
>   -H 'content-type: application/json' \
>   -d '{
>   "signature_name": "serving_default",
>   "inputs":
>     {
>        "input": { "b64": "/9j/4AAQSkZJRgABAQEAeAB4AAD/4QAiRXhpZgAATU0AKgAAAAgAAQESAAMAAAABAAEAAAAAAAD/2wBDAAIBAQIBAQICAgICAgICAwUDAwMDAwYEBAMFBwYHBwcGBwcICQsJCAgKCAcHCg0KCgsMDAwMBwkODw0MDgsMDAz/2wBDAQICAgMDAwYDAwYMCAcIDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAz/wAARCAA+AD4DASIAAhEBAxEB/8QAHwAAAQUBAQEBAQEAAAAAAAAAAAECAwQFBgcICQoL/8QAtRAAAgEDAwIEAwUFBAQAAAF9AQIDAAQRBRIhMUEGE1FhByJxFDKBkaEII0KxwRVS0fAkM2JyggkKFhcYGRolJicoKSo0NTY3ODk6Q0RFRkdISUpTVFVWV1hZWmNkZWZnaGlqc3R1dnd4eXqDhIWGh4iJipKTlJWWl5iZmqKjpKWmp6ipqrKztLW2t7i5usLDxMXGx8jJytLT1NXW19jZ2uHi4+Tl5ufo6erx8vP09fb3+Pn6/8QAHwEAAwEBAQEBAQEBAQAAAAAAAAECAwQFBgcICQoL/8QAtREAAgECBAQDBAcFBAQAAQJ3AAECAxEEBSExBhJBUQdhcRMiMoEIFEKRobHBCSMzUvAVYnLRChYkNOEl8RcYGRomJygpKjU2Nzg5OkNERUZHSElKU1RVVldYWVpjZGVmZ2hpanN0dXZ3eHl6goOEhYaHiImKkpOUlZaXmJmaoqOkpaanqKmqsrO0tba3uLm6wsPExcbHyMnK0tPU1dbX2Nna4uPk5ebn6Onq8vP09fb3+Pn6/9oADAMBAAIRAxEAPwD+f+iiigAoooxQAUUFSpwRgjgijHFABRRRQAUDrRQDg0Adb8FPg14i/aA+Lnh3wX4V0+TUvEXii+i0/T7ZASZZXYKOBk4HUnHABPQV+gf/AAV1/wCDdjxN/wAEsP2WPCvxKm8XWniq31DU00bW7ZYvK/s+4lR3gCf31IRgT2YCvVf+DPv9jCP4y/txeIfirqtu0mn/AAt04rYF0BU6hcgopOejLHvxz/FX6d/8F4f2APFX/BYrwjD8N/hd8aPAtjqPwxmfVNc8Ezyia6ur11xbvPIkha3AjdggdNhZm+bg4APzL/4NXP8Agmj8C/8AgodpXxmPxY8Lr4s1PwzLpgsYZrmSEQwz+cTIuwglt0RB9iPWvgv/AILBfAbwn+zH/wAFIfiv4H8DWbab4T0DVvs+m2rOX8iMRplcnk/Pu61+mX/Bqn8PfG37EP8AwV9+MXwR8faRLoXiq38MvbalZzSZ8t7eaCaKSMj5ZElimDowyGjcMDg5r87v+C7GoR6l/wAFZPjdLGwkVfEUsZKnIBAAI+vtQB8jUUUUAFKpww+vcUlFAH6If8Edf+C/viL/AIJF/CXxt4X0jwDo/jL/AISi7jvrN7y+NotlKqFWL7Iy0i9CFyPQEV84Rf8ABSr4waN+2f4g+PvhnxhqHhH4j+IdRm1Ca+02TCHzCB5DxvuSSEJgbZFZTjkGvn6t74Z+CG+JXxI0Dw7HcQ2ja5qNtpyzyHCQmaVI97ew3ZPsKAPvj/gnp+2b8XfhJ8SPiL+2J4n1zUPEmvxxJpc+rarKZbjUmLwiRVwynZGiwxbR+7HnRKUIIFfCvxm+KGtfHb4n+JPHHiCZrzWvFOpT6nfTY4M0zl2+gycAegr+qD9p/wD4Ia/s46Z+xX8Mfhv478YHwD8P/hmVv9Xnhu4LFvEs4Xa0k8j5bBeR24ycsuclQa/In/gvP+1r+yPP8CvCnwJ/Zb8K6TDY+F9VbUdT17T4CsdywQp5ZmfMlwxPzFi2B2FAH5U0UUUAFFFFAADUlrO9rcxyRyNFJGwZXVipQg5BBHIx61HRQB6b8cf2w/id+0rFZp498eeKPFUVjEIoYdQvnkjiUYACrnHYGvNSVx17cd6ZRQAUUUUAf//Z"
>      }
>    }
> }'
{
    "outputs": {
        "probability": 0.915982,
        "output": "T"
    }

**Comment by emedvedev** [comment]
I'm going to leave this issue open for a while in case someone wants to jump in, but as I've mentioned, I don't have the capacity to debug forks, unfortunately. Hopefully someone will be able to help or you'll be able to figure it out :)

**Comment by kspook** [comment]
@emedvedev , for your information, the exported model can't return Korean when I use 'stdin' in predict() at __main__.py to get image file.

Now I can't get number well when I changed argument style to get image.

**Comment by kspook** [comment]
@emedvedev, I think I found the reason.

all  the script I ran, was that I used 'python aocr/__main__.py ... ',
because the changed part was applied to 'aocr train/predict/...'

In this case, the output and saved_model was different.

How can I appy my changes with 'aocr train/predict/...'

**Comment by kspook** [comment]
@emedvedev,  I finally found out   the difference b/  saved_model and script is  bug of your code.


![image](https://user-images.githubusercontent.com/36400753/60235548-1fcc6d80-98e3-11e9-814c-f3e5ca7fbc3e.png)
